classes/decision/types/BackToPreviousInternalReviewRound.inc.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+/**
+ * @file classes/decision/types/BackToPreviousInternalReviewRound.inc.php
+ *
+ * Copyright (c) 2014-2022 Simon Fraser University
+ * Copyright (c) 2000-2022 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class decision
+ *
+ * @brief A decision to back out to previous internal review round from current round
+ */
+
+namespace PKP\decision\types;
+
+use APP\core\Application;
+use APP\decision\Decision;
+use APP\submission\Submission;
+use Illuminate\Validation\Validator;
+use PKP\context\Context;
+use PKP\db\DAORegistry;
+use PKP\decision\DecisionType;
+use PKP\decision\Steps;
+use PKP\decision\steps\Email;
+use PKP\decision\types\contracts\DecisionRetractable;
+use PKP\decision\types\traits\InInternalReviewRound;
+use PKP\decision\types\traits\NotifyAuthors;
+use PKP\decision\types\traits\NotifyReviewers;
+use PKP\decision\types\traits\withReviewRound;
+use PKP\mail\mailables\DecisionBackToPreviousInternalReviewRoundNotifyAuthor;
+use PKP\mail\mailables\DecisionBackToPreviousInternalReviewRoundNotifyReviewer;
+use PKP\submission\reviewAssignment\ReviewAssignmentDAO;
+use PKP\submission\reviewRound\ReviewRound;
+use PKP\user\User;
+
+class BackToPreviousInternalReviewRound extends DecisionType implements DecisionRetractable
```

Comment:
@NateWr  rather than extending, better to move the similar code into a trait . One question , when notifying authors or reviewers , do we need to specify the `review round stage(external or internal)`  . So rather than specifying `back to previous internal review round` or `back to previous external review round` , can we just say `Back to previous round` ? that way we can have a 2 mailables instead of 4 . 

So right now we have 
1. DecisionBackToPreviousInternalReviewRoundNotifyAuthor
2. DecisionBackToPreviousInternalReviewRoundNotifyReviewer
3. DecisionBackToPreviousExternalReviewRoundNotifyAuthor
4. DecisionBackToPreviousExternalReviewRoundNotifyReviewer

I want to reduce it to 
1. DecisionBackToPreviousReviewRoundNotifyAuthor
2. DecisionBackToPreviousReviewRoundNotifyReviewer

what you say ? this way a lot of code just can be moved to a trait say `ToPreviousRound` .

Reply:
Done .

Reply:
Done .